src/app/pages/dashboard/widgets/memory/widget-pool-name/widget-pool-name-settings/widget-pool-name-settings.component.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  ChangeDetectionStrategy, Component, OnInit,
+} from '@angular/core';
+import { Validators } from '@angular/forms';
+import { FormBuilder } from '@ngneat/reactive-forms';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { filter, map } from 'rxjs';
+import { getAllFormErrors } from 'app/modules/ix-forms/utils/get-form-errors.utils';
+import { WidgetResourcesService } from 'app/pages/dashboard/services/widget-resources.service';
+import { WidgetSettingsComponent } from 'app/pages/dashboard/types/widget-component.interface';
+import { WidgetSettingsRef } from 'app/pages/dashboard/types/widget-settings-ref.interface';
+import { WidgetPoolNameSettings } from 'app/pages/dashboard/widgets/memory/widget-pool-name/widget-pool-name.definition';
+
+@UntilDestroy()
+@Component({
+  selector: 'ix-widget-pool-name-settings',
+  templateUrl: './widget-pool-name-settings.component.html',
+  styleUrl: './widget-pool-name-settings.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class WidgetPoolNameSettingsComponent implements WidgetSettingsComponent<WidgetPoolNameSettings>, OnInit {
+  form = this.fb.group({
+    pool: [null as string, [Validators.required]],
+  });
+
+  protected poolOptions$ = this.resources.poolList$.pipe(
+    filter((state) => !!state.value && !state.isLoading),
+    map((state) => state.value),
+    map((pools) => (pools || []).map((result) => ({
```

Comment:
It looks like the two steps above are unnecessary. I'd assume `state.value` is producing an empty array since `pool.query` responds with an array. So, `state.value` can be excluded from the condition in `filter` and two `map` can be combined into a single one. Furthermore, there is an `idNameArrayToOptions` operator that can be used here.